### ISSUE TITLE
cpp/IceBox/configuration failure (macos debug)

### ISSUE THREAD

**Issue by bernardnormier** [zeroc-ice/ice#3016]
From https://github.com/zeroc-ice/ice/actions/runs/11599898627/job/32299105982

```
*** [31/340] Running cpp/IceBox/configuration tests ***
[ running client/server #1 test - 10/30/24 19:16:47 ]
- Config: tcp
(/Users/runner/work/ice/ice/cpp/bin/icebox --Ice.Default.Host=127.0.0.1 --Ice.Warn.Connections=1 --Ice.Default.Protocol=tcp --Ice.IPv6=0 --Ice.PrintStackTraces=1 --Ice.ThreadPool.Server.Size=1 --Ice.ThreadPool.Server.SizeMax=3 --Ice.ThreadPool.Server.SizeWarn=0 --Ice.PrintAdapterReady=1 --Ice.Config=/Users/runner/work/ice/ice/cpp/test/IceBox/configuration/config.icebox env={'DYLD_LIBRARY_PATH': 'build/macosx/shared'})
(/Users/runner/work/ice/ice/cpp/test/IceBox/configuration/build/macosx/shared/client --Ice.Default.Host=127.0.0.1 --Ice.Warn.Connections=1 --Ice.Default.Protocol=tcp --Ice.IPv6=0 --Ice.PrintStackTraces=1 env={'DYLD_LIBRARY_PATH': 'build/macosx/shared'})
testing service properties... ok
testing with shared communicator... ok
[ running client/server #1 test - 10/30/24 19:16:47 ]
- Config: tcp,compress,serialize
(/Users/runner/work/ice/ice/cpp/bin/icebox --Ice.Default.Host=127.0.0.1 --Ice.Warn.Connections=1 --Ice.Default.Protocol=tcp --Ice.Override.Compress=1 --Ice.ThreadPool.Server.Serialize=1 --Ice.IPv6=0 --Ice.PrintStackTraces=1 --Ice.ThreadPool.Server.Size=1 --Ice.ThreadPool.Server.SizeMax=3 --Ice.ThreadPool.Server.SizeWarn=0 --Ice.PrintAdapterReady=1 --Ice.Config=/Users/runner/work/ice/ice/cpp/test/IceBox/configuration/config.icebox env={'DYLD_LIBRARY_PATH': 'build/macosx/shared'})
(/Users/runner/work/ice/ice/cpp/test/IceBox/configuration/build/macosx/shared/client --Ice.Default.Host=127.0.0.1 --Ice.Warn.Connections=1 --Ice.Default.Protocol=tcp --Ice.Override.Compress=1 --Ice.ThreadPool.Server.Serialize=1 --Ice.IPv6=0 --Ice.PrintStackTraces=1 env={'DYLD_LIBRARY_PATH': 'build/macosx/shared'})
testing service properties... ok
testing with shared communicator... ok
libc++abi: terminating due to uncaught exception of type Ice::ObjectAdapterDeactivatedException: object adapter 'Ice.Admin' is deactivated
-! 10/30/24 19:16:47486 /Users/runner/work/ice/ice/cpp/test/IceBox/configuration/build/macosx/shared/client: warning: connection exception:
   src/Ice/StreamSocket.cpp:215 ::Ice::ConnectionLostException connection lost: recv() returned zero
   stack trace:
     0 Ice::Exception::Exception(char const*, int, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>) in libIce.38a0.dylib
     1 Ice::LocalException::LocalException(char const*, int, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>) in libIce.38a0.dylib
     2 Ice::SyscallException::SyscallException(char const*, int, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, int, std::__1::function<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> (int)>) in libIce.38a0.dylib
     3 Ice::SocketException::SocketException(char const*, int, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, int, std::__1::function<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> (int)>) in libIce.38a0.dylib
     4 Ice::ConnectionLostException::ConnectionLostException(char const*, int, int) in libIce.38a0.dylib
     5 Ice::ConnectionLostException::ConnectionLostException(char const*, int, int) in libIce.38a0.dylib
     6 IceInternal::StreamSocket::read(char*, unsigned long) in libIce.38a0.dylib
     7 IceInternal::StreamSocket::read(IceInternal::Buffer&) in libIce.38a0.dylib
     8 IceInternal::TcpTransceiver::read(IceInternal::Buffer&) in libIce.38a0.dylib
     9 IceInternal::IdleTimeoutTransceiverDecorator::read(IceInternal::Buffer&) in libIce.38a0.dylib
    10 Ice::ConnectionI::read(IceInternal::Buffer&) in libIce.38a0.dylib
    11 Ice::ConnectionI::message(IceInternal::ThreadPoolCurrent&) in libIce.38a0.dylib
    12 IceInternal::ThreadPool::run(std::__1::shared_ptr<IceInternal::ThreadPool::EventHandlerThread> const&) in libIce.38a0.dylib
    13 IceInternal::ThreadPool::EventHandlerThread::run() in libIce.38a0.dylib
    14 decltype(*std::declval<IceInternal::ThreadPool::EventHandlerThread*>().*std::declval<void (IceInternal::ThreadPool::EventHandlerThread::*)()>()()) std::__1::__invoke[abi:ne[180](https://github.com/zeroc-ice/ice/actions/runs/11599898627/job/32299105982#step:5:182)100]<void (IceInternal::ThreadPool::EventHandlerThread::*)(), IceInternal::ThreadPool::EventHandlerThread*, void>(void (IceInternal::ThreadPool::EventHandlerThread::*&&)(), IceInternal::ThreadPool::EventHandlerThread*&&) in libIce.38a0.dylib
    15 void std::__1::__thread_execute[abi:ne180100]<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct>>, void (IceInternal::ThreadPool::EventHandlerThread::*)(), IceInternal::ThreadPool::EventHandlerThread*, 2ul>(std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct>>, void (IceInternal::ThreadPool::EventHandlerThread::*)(), IceInternal::ThreadPool::EventHandlerThread*>&, std::__1::__tuple_indices<2ul>) in libIce.38a0.dylib
    16 void* std::__1::__thread_proxy[abi:ne180100]<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct>>, void (IceInternal::ThreadPool::EventHandlerThread::*)(), IceInternal::ThreadPool::EventHandlerThread*>>(void*) in libIce.38a0.dylib
    17 _pthread_start in libsystem_pthread.dylib
    18 thread_start in libsystem_pthread.dylib

```

**Comment by externl** [comment]
Do you think it's possible this could be related to my recent property changes?

**Comment by bernardnormier** [comment]
After looking at the stack trace attached to the build, it appears there are two bugs:
 - getLocator throws even though it's noexcept (fixed by #3028)
 - somehow we are shutting down IceBox as it's starting up (a test script issue?)
